kernel/src/common/deferred_call_mux.rs

```diff
@@ -0,0 +1,239 @@
+//! Hardware-independent kernel interface for deferred calls
+//!
+//! This allows any struct in the kernel which implements
+//! [DeferredCallMuxClient](crate::common::deferred_call_mux::DeferredCallMuxClient)
+//! to set and receive deferred calls.
+//!
+//! These are scheduled "interrupts" in the chip scheduler.
+//! They are especially important if some hardware doesn't
```

Comment:
We should remove the part about using this for hardware that doesn't support necessary interrupts, since we want that mechanism to be separate (and the responsibility of the chip implementation). Can we focus the comment of this module to just be on supporting asynchronous software operations (such as long running crypto operations)?

Reply:
Done. I hope my explanation for returning the call-stack up to the scheduler is clear enough. This is actually one of the primary use-cases for me, as my function call often happens in a `TakeCell`, and calling back to the client again tries to map that cell, failing of course.